src/init.cpp

```diff
@@ -1920,13 +1917,34 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     connOptions.nMaxOutboundLimit = nMaxOutboundLimit;
     connOptions.m_peer_connect_timeout = peer_connect_timeout;
 
-    for (const std::string& strBind : args.GetArgs("-bind")) {
-        CService addrBind;
-        if (!Lookup(strBind, addrBind, GetListenPort(), false)) {
-            return InitError(ResolveErrMsg("bind", strBind));
+    connOptions.listen_onion = args.GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION);
+
+    for (const std::string& bind_arg : args.GetArgs("-bind")) {
+        CService bind_addr;
+        const size_t index = bind_arg.rfind('=');
+        if (index == std::string::npos) {
+            if (Lookup(bind_arg, bind_addr, GetListenPort(), false)) {
+                connOptions.vBinds.push_back(bind_addr);
+                continue;
+            }
+        } else {
+            const std::string network_type = bind_arg.substr(index + 1);
+            if (connOptions.listen_onion && network_type == "onion") {
+                const std::string truncated_bind_arg = bind_arg.substr(0, index);
+                if (Lookup(truncated_bind_arg, bind_addr, BaseParams().OnionServiceTargetPort(), false)) {
+                    connOptions.onion_bind = bind_addr;
```

Comment:
> This would accept multiple =onion arguments and discard all but the last one.

That is done intentionally, leaving improving for follow ups.

Reply:
I think it is not good to introduce one behavior and change it later. Like "ignore all but last `-bind=addr:port=onion`" and later change it to something else because users may start to depend on the first one.

If supporting more than one `-bind=addr:port=onion` is undesired then what about emitting an error in that case?

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/19991#issuecomment-700831638).